File: TextosJapones/src/main/java/org/jisho/textosJapones/model/services/ComicInfoServices.java
package org.jisho.textosJapones.model.services;

import org.jisho.textosJapones.database.dao.ComicInfoDao;
import org.jisho.textosJapones.database.dao.DaoFactory;
import org.jisho.textosJapones.model.entities.comicinfo.ComicInfo;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

public class ComicInfoServices {

    private final ComicInfoDao comicInfoDao = DaoFactory.createComicInfoDao();

    public ComicInfo select(String comic, String linguagem) throws ExcessaoBd {
        return comicInfoDao.select(comic, linguagem);
    }

    public void save(ComicInfo comic) throws ExcessaoBd {
        ComicInfo saved = select(comic.getComic(), comic.getLanguageISO());
        if (saved == null || saved.getId() == null)
            comicInfoDao.insert(comic);
        else {
            comic.setId(saved.getId());
            comicInfoDao.update(comic);
        }
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/service/ComicInfoServices.kt
package br.com.fenix.processatexto.service

import br.com.fenix.processatexto.database.DaoFactory
import br.com.fenix.processatexto.database.dao.ComicInfoDao
import br.com.fenix.processatexto.model.entities.comicinfo.ComicInfo
import java.sql.SQLException
import java.util.*


class ComicInfoServices {
    private val comicInfoDao: ComicInfoDao = DaoFactory.createComicInfoDao()

    @Throws(SQLException::class)
    fun select(comic: String, linguagem: String): Optional<ComicInfo> = comicInfoDao.select(comic, linguagem)

    @Throws(SQLException::class)
    fun save(comic: ComicInfo) {
        val saved = select(comic.comic!!, comic.languageISO!!)
        if (saved.isEmpty || saved.get().getId() == null)
            comicInfoDao.insert(comic)
        else {
            comic.setId(saved.get().getId())
            comicInfoDao.update(comic)
        }
    }
}